src/v/features/feature_table.cc

```diff
@@ -608,8 +610,20 @@ feature_table::decode_version_fence(model::record_batch batch) {
 }
 
 void feature_table::set_original_version(cluster::cluster_version v) {
-    _original_version = v;
-    if (v != cluster::invalid_version) {
+    vlog(
+      featureslog.debug,
+      "Try setting original_version {} -> {} (active_version: {})",
+      _original_version,
+      v,
+      _active_version);
+
+    if (
+      _original_version == cluster::invalid_version
+      || (v != cluster::invalid_version && v >= _original_version)) {
+        _original_version = v;
+    }
+
+    if (v != cluster::invalid_version && v >= _original_version) {
```

Comment:
seems like the `v != cluster::invalid_version` checks are not necessary here, but I wanted to make intent very clear for this draft. Can remove later.

My biggest concern is whether the intended semantics are preserved here.

Reply:
Judging by https://buildkite.com/redpanda/redpanda/builds/42905#018c6f86-c120-47ad-a55d-e622a5ec357d, we've violated some assumption that was put in place deliberately. The assertion was put in place _before_ the introduction of legacy defaults (both by John), so perhaps it's the assertion that should be removed.